packages/tools/client-debugger/src/Registry.ts

```diff
@@ -45,6 +46,66 @@ export interface FluidClientDebuggerProps {
 	containerNickname?: string;
 }
 
+export interface DebuggerRegistryEvents extends IEvent {
+	/**
+	 * Emitted when a {@link FluidClientDebugger | clientDebugger} is registered.
+	 *
+	 * @eventProperty
+	 */
+	(event: "debuggerRegistered", listener: (containerId: string) => void): void;
+
+	/**
+	 * Emitted when a {@link FluidClientDebugger | clientDebugger} is closed.
+	 *
+	 * @eventProperty
+	 */
+	(event: "debuggerClosed", listener: (containerId: string) => void): void;
+}
+
+export class DebuggerRegistry extends IEventProvider<DebuggerRegistryEvents> {
+	private readonly registeredDebuggers: Map<string, FluidClientDebugger> = new Map();
+
+	constructor() {
```

Comment:
Nit: 
```suggestion
	public constructor() {
```

I would also recommend running `npm run lint` from inside the package directory to make sure we're satisfying our linter requirements.